src/scene.cpp

```diff
@@ -26,6 +26,13 @@
 #include "audio.h"
 #include "transition.h"
 
+#ifndef NDEBUG
```

Comment:
Not sure if this is always defined when using autotools/cmake. It should for release builds (disabling asserts), but would need a quick test.

Reply:
for me it wouldnt ever matter if this is enabled in general, this check doesn't really hurt.

Reply:
We use `assert()` in other places. So if `NDEBUG` is misconfigured in some cases, we already have a bug.

Reply:
`NDEBUG` is not enabled for autotools builds, but cmake enables it for release builds.
Our other Makefiles and MSVC solution also explicitly do this.
So is fine to have right now and maybe change/fix the autotools configuration later.